packages/build/src/plugins_core/functions/index.js

```diff
@@ -63,6 +68,22 @@ const getZisiParameters = ({ buildDir, featureFlags, functionsConfig, functionsD
   return { basePath: buildDir, config, manifest, featureFlags: zisiFeatureFlags }
 }
 
+// TODO: is this the best place for this logic? should this live elsewhere, e.g. in its own plugin?
+const writeToScheduleFile = async (zisiResult) => {
+  const schedule = zisiResult
+    .filter(({ config }) => Boolean(config.schedule))
+    .map(({ name, config }) => ({ route: `.netlify/functions/${name}`, schedule: config.schedule }))
+
+  if (schedule.length === 0) {
+    return
+  }
+
+  const scheduleFile = JSON.stringify(schedule, null, 2)
+
+  // TODO: what's the exact path where we should write? somewhere in .netlify? in the root dir?
```

Comment:
@eduardoboucas I think the idea was to emit `_schedule` into the root directory, similar to (I think) `_headers` or `_redirects`. Just wanted to verify, is this the best location? Or would something like `.netlify/_schedule` work better?

Reply:
If we want to follow `_headers` and `_redirects`, I think we'd want to put it inside the publish directory.

@ehmicky do you see any issues with that?

Reply:
We should distinguish between the root directory and the build directory (which is used by this PR currently). The build directory can be either the root directory or (if one is used) the `base` directory.
 
The publish directory is used for `_headers` and `_redirects`, not the build directory nor root directory, as mentioned by @eduardoboucas.

I am guessing `_schedule` is being used by the API at the end of the build, after Netlify Build finished? At the end of the build, the buildbot uploaded the publish directory's contents to the API. The rest of the container files are discarded. So, the publish directory would be better there. However, we would need to make sure `_schedule` is then using some similar logic as `_headers` and `_redirects` so they do not get served to end users.

Have we considered making an API request to pass this information between Netlify Build and the API, instead of relying on the filesystem? This seems more proper to me, especially considering how small the payload is, although would require creating an API endpoint for it. Netlify Build already does a few API calls during the build like:

https://github.com/netlify/build/blob/36a1d3b135fa31d7a4569bc9d2e0dc87906a05bc/packages/build/src/error/cancel.js#L9

https://github.com/netlify/build/blob/36a1d3b135fa31d7a4569bc9d2e0dc87906a05bc/packages/build/src/plugins/pinned_version.js#L113

https://github.com/netlify/build/blob/36a1d3b135fa31d7a4569bc9d2e0dc87906a05bc/packages/build/src/status/report.js#L120

And (local builds only):

https://github.com/netlify/build/blob/36a1d3b135fa31d7a4569bc9d2e0dc87906a05bc/packages/config/src/api/site_info.js#L32

https://github.com/netlify/build/blob/36a1d3b135fa31d7a4569bc9d2e0dc87906a05bc/packages/config/src/api/site_info.js#L41

https://github.com/netlify/build/blob/36a1d3b135fa31d7a4569bc9d2e0dc87906a05bc/packages/config/src/api/site_info.js#L54

Reply:
> It seems to me the publish directory is mostly meant as a way for users to pass information from the build to our API, not for our own code.

I imagine that @netlify/integrations would be happy if they could amend to `_schedule` inside of a build plugin, also I can see frameworks outputting directly into it in cases where our `cron` detection doesn't work. For that reason it'd be good if it was user-writeable. Solving this via API calls would make it a lot harder for build plugins / frameworks.

Reply:
(cc @vbrown608)

Reply:
Yes, we would very much like to be able to create these in plugins. Not too bother about how (though manually hitting an API isn't preferred, as we don't do that for anything else)

Reply:
I don't think we should encourage anyone — end users or plugins — to write directly to this file. That expectation would place a maintainability burden on us, as we wouldn't be able to easily change the format or the contents of what is supposed to be an internal communication channel between our build and deployment stages.

If plugins wish to modify scheduling configurations, they can do so by mutating the configuration object. That's the pattern we've been using for other settings.

```js
module.exports = {
  onInit({ netlifyConfig }) {
    netlifyConfig.functions.foo.schedule = '@daily'
  }
}
```

Reply:
That's a good point. Should we even name it `_schedule` in that case? @eduardoboucas you mentioned a file called `netlify-toolchain` somewhere else, which sounded very much like the internal communication channel we're looking for. Would this be the right location for our schedule information, too? (a quick search didn't bring up anything on `netlify-toolchain`, so I don't know what else it's used for)

Reply:
That's a good point. Should we even name it _schedule in that case? @eduardoboucas you mentioned a file called netlify-toolchain somewhere else, which sounds like the internal communication channel we're looking for. Would this be the right location for our schedule information, too? (a quick search didn't bring up anything on netlify-toolchain, so I don't know what else it's used for)

Reply:
Yes, that was my initial suggestion. That file is currently used by zip-it-and-ship-it to tell the receiving party (Bitballoon) about the runtime used by the function. The file is not always present (although we could change that) and I don't know how convenient it would be, from a backend perspective, to read additional data from it. Perhaps @vbrown608 can advise on this.

At the end of the day, I don't have a strong preference about the specific location of the file, as long as we can make it git-ignored by default (hence the `.netlify/` suggestion) and frame it as something that shouldn't be interacted with by users or plugins.

Reply:
Mutating `netlifyConfig` is perfect for me

Reply:
> I do wonder whether making an API call from Netlify Build would avoid some of those issues

The information in `_schedule` will be an important part of the deploy. It's not some meta information, but integral to the behaviour of that deploy.
My understanding is that by writing the schedule information to a file (aka: producing a "tangible" output), we convey that this is indeed *part* of the deployment, and not some meta information.
Uploading the schedule through a separate API call (to me) conveys "this is meta information", and not so much that it's an integral part of the deployment.

This is an abstract argument at first, but I can imagine that it'll show e.g. in debugging: if we embed `_schedule` into the same zip file that's uploaded to Bitballoon and that also contains `_redirects` and `_headers`, then we can continue to examine weird-behaving builds solely based on that zip file.

The API calls that you listed @ehmicky sound like they're meant for interacting with the deploy process, and not about uploading meta information about the deploy itself; so I don't think they're a good guideline for this decision (for the reasons above).